tests/test_intervaldata.py

```diff
@@ -0,0 +1,196 @@
+import random
+
+import numpy as np
+import pytest
+
+from eitprocessing.datahandling.intervaldata import Interval, IntervalData
+
+
+@pytest.fixture()
+def intervaldata1():
+    """Creates an IntervalData object with n sequential intervals lasting 1 second each."""
+    n = random.randint(50, 150)
+    return IntervalData(
+        label="intervaldata1",
+        name="IntervalData 1",
+        unit=None,
+        category="dummy",
+        intervals=list(zip(range(n), range(1, n + 1), strict=True)),
+        default_partial_inclusion=True,
+    )
+
+
+@pytest.fixture()
+def intervaldata2():
+    """Creates an IntervalData object with n sequential intervals lasting 1 second each."""
+    n = random.randint(50, 150)
+    return IntervalData(
+        label="intervaldata1",
+        name="IntervalData 2",
+        unit=None,
+        category="dummy",
+        intervals=list(zip(range(n), range(1, n + 1), strict=True)),
+        default_partial_inclusion=False,
+    )
+
+
+@pytest.fixture()
+def intervaldata3():
+    """Creates an IntervalData object with n sequential intervals lasting 1 second each, with values."""
+    n = random.randint(50, 150)
+    return IntervalData(
+        label="intervaldata1",
+        name="IntervalData 2",
+        unit=None,
+        category="dummy",
+        intervals=list(zip(range(n), range(1, n + 1), strict=True)),
+        default_partial_inclusion=False,
+        values=[random.random() for _ in range(n)],
+    )
+
+
+@pytest.fixture()
+def intervaldata4():
+    """Creates an IntervalData object with n sequential intervals lasting 1 second each, with values."""
+    n = random.randint(50, 150)
+    return IntervalData(
+        label="intervaldata1",
+        name="IntervalData 2",
+        unit=None,
+        category="dummy",
+        intervals=list(zip(range(n), range(1, n + 1), strict=True)),
+        default_partial_inclusion=False,
+        values=np.array([random.random() for _ in range(n)]),
+    )
+
+
+def test_post_init(intervaldata1: IntervalData) -> None:
+    assert isinstance(intervaldata1.intervals, list)
+    assert all(isinstance(interval, Interval) for interval in intervaldata1.intervals)
+
+
+def test_len(intervaldata1: IntervalData) -> None:
+    assert len(intervaldata1) == len(intervaldata1.intervals)
+
+
+def test_has_values(
+    intervaldata1: IntervalData,
+    intervaldata2: IntervalData,
+    intervaldata3: IntervalData,
+    intervaldata4: IntervalData,
+) -> None:
+    assert not intervaldata1.has_values
+    intervaldata1.values = []
+    assert intervaldata1.has_values
+    intervaldata1.values = None
+
+    assert not intervaldata2.has_values
+    assert intervaldata3.has_values
+    assert intervaldata4.has_values
+
+
+def test_index_slicing(intervaldata1: IntervalData) -> None:
+    _sliced_copy = intervaldata1._sliced_copy(0, 10, newlabel="sliced_copy")  # noqa: SLF001
```

Comment:
why did we make `_sliced_copy` a private method? I think it would be nice to have as a public method, in case users want to add a custom label.
If we want to change this, I suggest to do it in a separate PR.

Reply:
Good question. I see no reason to keep it private. I created #212.